Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Ride.hs

```diff
@@ -110,22 +112,40 @@ getRideStatus ::
   m GetRideStatusResp
 getRideStatus rideId personId = withLogTag ("personId-" <> personId.getId) do
   ride <- runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
+  fromLocation <- QTripLocation.findById ride.fromLocationId >>= fromMaybeM (InternalError ride.id.getId)
   mbPos <-
     if ride.status == COMPLETED || ride.status == CANCELLED
       then return Nothing
       else Just <$> CallBPP.callGetDriverLocation ride
   booking <- runInReplica $ QRB.findById ride.bookingId >>= fromMaybeM (BookingDoesNotExist ride.bookingId.getId)
   rider <- runInReplica $ QP.findById booking.riderId >>= fromMaybeM (PersonNotFound booking.riderId.getId)
   decRider <- decrypt rider
-  return $
-    GetRideStatusResp
-      { fromLocation = makeBookingLocationAPIEntity booking.fromLocation,
-        toLocation = case booking.bookingDetails of
-          DB.OneWayDetails details -> Just $ makeBookingLocationAPIEntity details.toLocation
-          DB.RentalDetails _ -> Nothing
-          DB.OneWaySpecialZoneDetails details -> Just $ makeBookingLocationAPIEntity details.toLocation
-          DB.DriverOfferDetails details -> Just $ makeBookingLocationAPIEntity details.toLocation,
-        ride = makeRideAPIEntity ride,
-        customer = SPerson.makePersonAPIEntity decRider,
-        driverPosition = mbPos <&> (.currPoint)
-      }
+
+  case ride.toLocationId of
+    Just toLocationId -> do
+      toLocation <- QTripLocation.findById toLocationId >>= fromMaybeM (InternalError ride.id.getId)
```

Comment:
> You don't have to fail here if `toLocation` is not present. You can take it as a `Maybe` value and map it over `makeTripLocationAPIEntity`. This will eliminate case match as well

If there is some id, which doesn't exist in the table -> It's clear DB internal error to me IMO

Reply:
In Rental details if toLocation is Just then also we have to pass nothing as toLocation does not exist for it so we can remove this error which is thrown but we have to do case match there somehow